s3prl/downstream/diarization/dataset.py

```diff
@@ -286,8 +286,15 @@ def _load_segments_rechash(self, segments_file):
 
     def _load_wav_scp(self, wav_scp_file):
         """Return dictionary { rec: wav_rxfilename }."""
-        lines = [line.strip().split(None, 1) for line in open(wav_scp_file)]
-        return {x[0]: x[1] for x in lines}
+        if os.path.exists(wav_scp_file):
```

Comment:
this change was added by @albertvillanova to simplify the file loading on our end. he can provide more details if you need them :)

Reply:
Yes! I understand the change, but I am curious that is the original scp file format not work for your infra? I think it will be nice to follow the current file structure, since all the

- scp
- segments
- recor2dur
- spk2uttr
- uttr2spk

These files under the `train/`, `dev/`, `test/` are quite standard as the Kaldi toolkit's file format which we followed in this task, and I guess for hidden set Jiatong (who is in charge of diarization) will stick to the same convention.

Do you think this scp style of wav loading will cause complication for your server? Please feel free to let me know :)

@lewis @albertvillanova

Reply:
Hi @leo19941227 and @lewtun,

First, let's put this decision in context: I was integrating the "sd" (diarization) dataset into Hugging Face Datasets; see huggingface/datasets/pull/2661. The aim of this is that end users do not have to run any script to prepare the data (as described here: https://github.com/s3prl/s3prl/tree/master/s3prl/downstream#prepare-data-8), but just load the dataset with:
```python
from datasets import load_dataset

dataset = load_dataset("superb", "sd")
```

In the backward, `datasets` downloads some files and generates the corresponding dataset (and caches it in a single file, in Apache Arrow format). Each item in the dataset has a field named "file" with the absolute path to the audio file.

However, this absolute path is user-dependent: it is generated while generating the dataset, depending on the path of the datasets cache directory.

We cannot provide the file "wav.scp" in advance, as we do not know the paths to the audio files before the user downloads the files to their cache directory. And once the files are downloaded, we do not need "wav.scp" either, as we already know the cache dir and the relative path to the audio files. Therefore, we generate the absolute path to the audio files while generating the dataset, and assign it to the "file" dataset field.

I hope this explanation is clear. And do not hesitate to ask for further details.

Reply:
@lewtun @leo19941227 

Yes, but for running fine-tuning experiments with `s3prl`, I regenerated the content in `/app/data/Libri2Mix` into an analog structure as if it was downloaded using `datasets`: 

I mean, if you look inside e.g. `/app/data/Libri2Mix/train` you can check that:
- there is no "wav.scp" file
- it contains a sub-direcotry called "wav" which contains all the audio files

This is very convenient because you could move the `Libri2Mix` data directory from `/data/.` to `/app/data/.` without having to modify the content of the "wav.scp" file.

Otherwise, the "wav.scp" file would have contained entries with absolute paths to the audios like:
```
file1 /data/Libri2Mix/train/wav/file1.wav
file2 /data/Libri2Mix/train/wav/file2.wav
...
```
and you would have needed to change all entries to:
```
file1 /app/data/Libri2Mix/train/wav/file1.wav
file2 /app/data/Libri2Mix/train/wav/file2.wav
...
```

Reply:
Cool! I can understand the concern. Just for a correctness check I think @ftshijt can help to review this change :)

Reply:
I think the refactoring is very cool! The wavscp style is just trying to align with Kaldi and ESPNet on diarization task, so as for easier importing (for the case if we want to include more data). If the user only needs to fine-tune the set on our default dataset, we don't necessarily need to generate them. I like the idea of keeping both cache and wavscp style! Many thanks to your effort @albertvillanova